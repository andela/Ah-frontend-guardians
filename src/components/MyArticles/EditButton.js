import React, { Component } from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";

export class EditButton extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const url = `article/${this.props.slug}/edit`;
    return (
      <div>
        <form>
          <NavLink to={url} article={this.props.article}>
            <input
              type="submit"
              className="btn btn-edit"
              value="Edit Article"/>
          </NavLink>
        </form>
      </div>
    );
  }
}

export const mapStateToProps = state => {
  return {
    edit_article: state.articleReducer.edit_article
  };
};

export default connect(mapStateToProps)(EditButton);
