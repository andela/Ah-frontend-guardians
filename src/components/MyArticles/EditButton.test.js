import React from 'react';
import { EditButton, mapStateToProps } from './EditButton';
import { shallow } from "enzyme";

const props = {
    slug: 'code',
    artocle: {title: "title"}
}

describe("Delete Button", () => {
  const wrapper = shallow( <EditButton {...props}  />);

  it("should render the component ", () => {
    
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should return the right props', () => {
    const mockedState = {
        articleReducer:{edit_article: {"slug":"slug"}}
    }
    const state = mapStateToProps(mockedState);
    expect(state.edit_article).toEqual({"slug":"slug"});
  });
});