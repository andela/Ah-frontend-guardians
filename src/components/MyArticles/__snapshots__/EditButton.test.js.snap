// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Button should render the component  1`] = `ShallowWrapper {}`;
